code/game/machinery/newscaster.dm

```diff
@@ -874,9 +874,8 @@ GLOBAL_LIST_EMPTY(allCasters)
 		say("Breaking news from [channel]!")
 		alert ++
 		update_icon()
-		spawn(alert_delay)
-			alert --
-			update_icon()
+		addtimer(VARSET_CALLBACK(src, alert, alert-1), alert_delay)
```

Comment:
this won't subtract 1 from alert after alert_delay. It will set alert to (CURRENT)alert - 1 after alert_delay